core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2168,7 +2168,17 @@ private[spark] object Utils extends Logging {
     // We need to filter out null values here because dumpAllThreads() may return null array
     // elements for threads that are dead / don't exist.
     val threadInfos = ManagementFactory.getThreadMXBean.dumpAllThreads(true, true).filter(_ != null)
-    threadInfos.sortBy(_.getThreadId).map(threadInfoToThreadStackTrace)
```

Comment:
`sortBy(_.getThreadId)` just disappeared. Is it intentional?

Reply:
Yes, it was. There is a new sortWith right after this line. Of course if two threads has the same name then earlier the one with the smaller thread ID was before (assuming sortWith is stable). I can add this extra conditions if it makes sense, @squito what do you think (how often we have threads with the same name)?

Reply:
Keeping old behaviour is a good idea this case.